secure/storage/vault/src/lib.rs

```diff
@@ -369,7 +371,13 @@ impl Client {
 
         // Trim keys if too many versions exist
         if (max_version - min_version) >= MAX_NUM_KEY_VERSIONS {
-            let min_available_version = max_version - MAX_NUM_KEY_VERSIONS + 1;
+            // let min_available_version = max_version - MAX_NUM_KEY_VERSIONS + 1;
+            let min_available_version = max_version
+                .checked_sub(MAX_NUM_KEY_VERSIONS)
+                .and_then(|n| n.checked_add(1))
+                .ok_or_else(|| {
+                    Error::OverflowError(String::from("trim_key_versions::min_available_version"))
```

Comment:
nit: you could also use `into` to do the string conversion for you. It's shorter: https://doc.rust-lang.org/std/convert/trait.Into.html, e.g.,:

`Error::OverflowError("trim_key_versions::min_available_version".into())`